File: Source/Back-end/src/main/java/org/elis/manutenzione/dto/response/MessaggioDTO.java
package org.elis.manutenzione.dto.response;

import lombok.Data;
import org.elis.manutenzione.model._enum.TipoMessaggioInviato;

/**
 * DTO che rappresenta un messaggio.
 */
@Data
public class MessaggioDTO {

	/**
	 * Identificativo del messaggio.
	 */
	private long id;

	/**
	 * Lo username del mittente del messaggio.
	 */
	private String mittente;

	/**
	 * Il nome del mittente del messaggio.
	 */
	private String nomeMittente;

	/**
	 * Il cognome del mittente del messaggio.
	 */
	private String cognomeMittente;

	/**
	 * Il testo del messaggio.
	 */
	private String testo;

	/**
	 * La data di invio del messaggio.
	 */
	private String data;


	/**
	 * Identificativo del messaggio di riferimento.
	 */
	private long idRiferimentoMessaggio;

	/**
	 * Il tipo del messaggio. Immagine o testo
	 */
	private TipoMessaggioInviato tipoMessaggio;
}

File: Source/Front-end/front-end/src/app/model/response/messaggio-dto.ts
import {TipoMessaggioInviato} from "../models/tipo-messaggio-inviato";
import {TipoMessaggio} from "../models/tipo-messaggio";

export class MessaggioDTO {
  id: number;
  mittente: string;
  nomeMittente: string;
  cognomeMittente: string;
  testo: string;
  data: Date;
  idRiferimentoMessaggio: number;
  tipoMessaggio: TipoMessaggioInviato

  constructor(mittente: string, testo: string, data: Date, tipoMessaggio: TipoMessaggioInviato, id?: number, idRiferimentoMessaggio?: number) {
    this.id = id ? id : 0;
    this.mittente = mittente;
    this.nomeMittente = "";
    this.cognomeMittente = "";
    this.testo = testo;
    this.data = data;
    this.idRiferimentoMessaggio = idRiferimentoMessaggio ? idRiferimentoMessaggio : 0;
    this.tipoMessaggio = tipoMessaggio
  }
}
